moveit_ros/hybrid_planning/global_planner/global_planner_plugins/src/moveit_planning_pipeline.cpp

```diff
@@ -97,45 +100,101 @@ moveit_msgs::msg::MotionPlanResponse MoveItPlanningPipeline::plan(
+          global_goal_handle->get_goal()->motion_sequence.items.at(waypoint_idx).req;
+
+      trajectory_msgs::msg::JointTrajectoryPoint traj_point;
+      for (const moveit_msgs::msg::JointConstraint& joint_constraint :
+           waypoint_req.goal_constraints.at(0).joint_constraints)
+      {
+        traj_point.positions.push_back(joint_constraint.position);
+      }
+
+      response.trajectory.joint_trajectory.points.push_back(traj_point);
+    }
+    return response;
   }
-  auto motion_plan_req = (global_goal_handle->get_goal())->motion_sequence.items[0].req;
-
-  // Set parameters required by the planning component
-  moveit_cpp::PlanningComponent::PlanRequestParameters plan_params;
-  plan_params.planner_id = motion_plan_req.planner_id;
-  plan_params.planning_pipeline = motion_plan_req.pipeline_id;
-  plan_params.planning_attempts = motion_plan_req.num_planning_attempts;
-  plan_params.planning_time = motion_plan_req.allowed_planning_time;
-  plan_params.max_velocity_scaling_factor = motion_plan_req.max_velocity_scaling_factor;
-  plan_params.max_acceleration_scaling_factor = motion_plan_req.max_acceleration_scaling_factor;
-
-  // Create planning component
-  auto planning_components = std::make_shared<moveit_cpp::PlanningComponent>(motion_plan_req.group_name, moveit_cpp_);
-
-  // Copy goal constraint into planning component
-  planning_components->setGoal(motion_plan_req.goal_constraints);
-
-  // Plan motion
-  auto plan_solution = planning_components->plan(plan_params);
-  if (plan_solution.error_code != moveit_msgs::msg::MoveItErrorCodes::SUCCESS)
+
+  // Else, there is only one waypoint.
+  else
   {
+    if ((global_goal_handle->get_goal())->motion_sequence.items.size() > 1)
+    {
+      RCLCPP_WARN(LOGGER, "Global planner received motion sequence request with more than one item but the "
+                          "'global_traj_pass_through' parameter is false. Just using the first item as global "
+                          "planning goal!");
+    }
+    auto motion_plan_req = global_goal_handle->get_goal()->motion_sequence.items[0].req;
+
+    // Set parameters required by the planning component
+    moveit_cpp::PlanningComponent::PlanRequestParameters plan_params;
+    plan_params.planner_id = motion_plan_req.planner_id;
+    plan_params.planning_pipeline = motion_plan_req.pipeline_id;
+    plan_params.planning_attempts = motion_plan_req.num_planning_attempts;
+    plan_params.planning_time = motion_plan_req.allowed_planning_time;
+    plan_params.max_velocity_scaling_factor = motion_plan_req.max_velocity_scaling_factor;
+    plan_params.max_acceleration_scaling_factor = motion_plan_req.max_acceleration_scaling_factor;
+
+    // Create planning component
+    auto planning_components = std::make_shared<moveit_cpp::PlanningComponent>(motion_plan_req.group_name, moveit_cpp_);
+
+    // Copy goal constraint into planning component
+    planning_components->setGoal(motion_plan_req.goal_constraints);
```

Comment:
I just realized that this ignores path constraints, so I would suggest to add this:
```suggestion
    // Copy constraints into planning component
    planning_components->setGoal(motion_plan_req.goal_constraints);
    planning_components->setPathConstraints(motion_plan_req.path_constraints);
```